hypothesis-python/src/hypothesis/internal/conjecture/data.py

```diff
@@ -2255,17 +2291,19 @@ def draw_string(
 
     def draw_bytes(
         self,
-        # TODO move to min_size and max_size here.
-        size: int,
+        min_size: int,
+        max_size: Optional[int],
         *,
         forced: Optional[bytes] = None,
         fake_forced: bool = False,
         observe: bool = True,
     ) -> bytes:
-        assert forced is None or len(forced) == size
-        assert size >= 0
+        assert forced is None or min_size <= len(forced)
+        assert min_size >= 0
```

Comment:
Add assertions for `max_size` too?

Reply:
semi-intentional because `max_size` gets checked in the provider after we handle the `max_size is None` case, like strings. This'll become cleaner when we get rid of that case though